bin/migrate_to_single_table.py

```diff
@@ -0,0 +1,268 @@
+"""
+
+"""
+import argparse
+import json
+import config_helper
+from cdislogging import get_logger
+from sqlalchemy import create_engine, MetaData, Table, Column, Integer, String, DateTime
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker
+import time
+import random
+import re
+
+import cProfile
+
+from indexd.index.drivers.alchemy import (
+    IndexRecord,
+    IndexRecordAuthz,
+    BaseVersion,
+    IndexRecordAlias,
+    IndexRecordUrl,
+    IndexRecordACE,
+    IndexRecordMetadata,
+    IndexRecordUrlMetadata,
+    IndexRecordHash,
+)
+from indexd.index.drivers.single_table_alchemy import Record
+
+APP_NAME = "indexd"
+
+logger = get_logger("migrate_single_table", log_level="debug")
+
+
+def load_json(file_name):
+    return config_helper.load_json(file_name, APP_NAME)
+
+
+def main():
+    migrator = IndexRecordMigrator()
+    migrator.index_record_to_new_table()
+    return
+
+
+class IndexRecordMigrator:
+    def __init__(self, conf_data=None):
+        self.logger = get_logger("migrate_single_table", log_level="debug")
+
+        if conf_data:
+            with open(conf_data, "r") as reader:
+                conf_data = json.load(reader)
+        else:
+            conf_data = load_json("creds.json")
+
+        usr = conf_data.get("db_username", "{{db_username}}")
+        db = conf_data.get("db_database", "{{db_database}}")
+        psw = conf_data.get("db_password", "{{db_password}}")
+        pghost = conf_data.get("db_host", "{{db_host}}")
+        pgport = 5432
+
+        engine = create_engine(
+            f"postgresql+psycopg2://{usr}:{psw}@{pghost}:{pgport}/{db}"
+        )
+
+        Base = declarative_base()
+        Base.metadata.create_all(engine)
+        Session = sessionmaker(bind=engine)
+
+        self.session = Session()
+
+    def index_record_to_new_table(self, batch_size=1000, retry_limit=4):
```

Comment:
I imagine this batch_size could be much larger. Where's the job that will run this? How much RAM do you plan on providing? Indexd records text is pretty darn small. Maybe 1000 characters on average? 1000 chars / record, 1 char = 1 byte in ASCII, 1kb per record. so 1000 records = 1MB. If we have 1 GBs of usable RAM, we could probably go up to like 10000 records.

Reply:
I'm not sure we _should_ do that, but if we can reliably, it might be faster (reducing the total number of queries and commits to the db). Would outputting to a file and using /COPY be faster? Don't you guys use /COPY for something already? Did you consider this?

Reply:
I did consider this. I think there's a couple of things here:
1. We had a conversation about whether to use sql+bash script vs python. We went with python because it would be easier for any gen3 user to use it. 
2. should we do python + /copy? My concern with this is that this would blow up for something like dcf where we have about 56mil records. I think we'd use up a lot of memory. We could do it in batches though 🤔

Reply:
Yeah, it's been a while since we've had the design conversations around the options here. Python makes sense, - we're already so close with this. I was just thinking out loud and not remembering previous conversations. Let's not shift gears at this point. I am curious if you were able to tune these numbers and avoid using OFFSET and if that helped